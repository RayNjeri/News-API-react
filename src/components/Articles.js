import React from 'react';
import Axios from 'axios';

export default class Articles extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      articles: [],
    };
  }

  componentDidMount() {
    Axios.get('https://newsapi.org/v1/articles?source=the-next-web&sortBy=latest&apiKey=3e94e65c70b047a3971837eb342fff8b')
      .then((result) => {
        console.log('getting result: ', result);
        this.setState({
          articles: result.data.articles
        });
      });
  }

  render() {
    return (
      < div className="articles">
        <h3>Articles</h3>
        {this.state.articles.map((articles, index) => (
          <div className="card col-xs-12 col-sm-6 col-md-4 m-2" key={index}>
            <div className="card-block">
              <img className="card-title" src={articles.urlToImage} style={{ width: '200', height: '200' }} />
              <a className="card-text" href={articles.url} target="_blank">{articles.title}</a>
            </div>
          </div>
        ))
        }
      </div >
    );
  }
};

