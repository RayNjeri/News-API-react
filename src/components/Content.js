import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Sources from './Sources'

export default class Content extends Component {
    render() {
        return (
            <div className="content">
                <span className="sources"><Sources /></span>
                <span>
                    <div className="filter">filter </div>
                    <div className="subContent">subContent</div>
                </span>
            </div>
        );
    }
}
