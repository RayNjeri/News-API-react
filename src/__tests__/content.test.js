import React from 'react';
import shallowToJSON from 'enzyme-to-json';
import { mount, shallow, wrapper } from 'enzyme';
import Content from '../components/Content.js';
import Articles from '../components/Articles';

describe('Content component', () => {
  it('Should have all the contents information', () => {
    const contents = shallow(<Content />);
    const tree = shallowToJSON(contents);
    expect(tree).toMatchSnapshot();
  });

   it('renders props correctly', () => {
    const wrapper = shallow(<Content name="renderArticles" />);
    expect(wrapper.instance().props.name).toBe('renderArticles');
  });

  it('should display Articles', () => {
    let mockDisplay = jest.fn();
    const wrapper = mount(<Content renderArticles={mockDisplay} />);
    expect(wrapper).toBeDefined();
  });

   it('check for presence of div elements', () => {
    const wrapper = shallow(<div></div>);
    expect(wrapper.find('div').node).toBeDefined();
  });
});    
