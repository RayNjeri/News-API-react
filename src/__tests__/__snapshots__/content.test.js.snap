// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component Should have all the contents information 1`] = `
<div
  className="content"
>
  <div
    className="row"
  >
    <div
      className="col-md-2"
    >
      <Sources />
    </div>
    <div
      className="col-md-8"
    >
      <Articles />
    </div>
    <div
      className="col-md-2"
    >
      <SourcesSearch />
    </div>
  </div>
</div>
`;
